sopht_mpi/utils/mpi_utils_3d.py

```diff
@@ -205,67 +200,98 @@ def exchange_init(self, local_field, mpi_construct):
         # Along Y: send to next block, receive from previous block
-        self.comm_requests[6] = mpi_construct.grid.Isend(
-            (local_field, self.send_next_along_y_type),
-            dest=mpi_construct.next_grid_along[y_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Isend(
+                (local_field, self.send_next_along_y_type),
+                dest=self.mpi_construct.next_grid_along[y_axis],
+            )
         )
-        self.comm_requests[7] = mpi_construct.grid.Irecv(
-            (local_field, self.recv_previous_along_y_type),
-            source=mpi_construct.previous_grid_along[y_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Irecv(
+                (local_field, self.recv_previous_along_y_type),
+                source=self.mpi_construct.previous_grid_along[y_axis],
+            )
         )
 
         # Along Z: send to previous block, receive from next block
-        self.comm_requests[8] = mpi_construct.grid.Isend(
-            (local_field, self.send_previous_along_z_type),
-            dest=mpi_construct.previous_grid_along[z_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Isend(
+                (local_field, self.send_previous_along_z_type),
+                dest=self.mpi_construct.previous_grid_along[z_axis],
+            )
         )
-        self.comm_requests[9] = mpi_construct.grid.Irecv(
-            (local_field, self.recv_next_along_z_type),
-            source=mpi_construct.next_grid_along[z_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Irecv(
+                (local_field, self.recv_next_along_z_type),
+                source=self.mpi_construct.next_grid_along[z_axis],
+            )
         )
         # Along Z: send to next block, receive from previous block
-        self.comm_requests[10] = mpi_construct.grid.Isend(
-            (local_field, self.send_next_along_z_type),
-            dest=mpi_construct.next_grid_along[z_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Isend(
+                (local_field, self.send_next_along_z_type),
+                dest=self.mpi_construct.next_grid_along[z_axis],
+            )
         )
-        self.comm_requests[11] = mpi_construct.grid.Irecv(
-            (local_field, self.recv_previous_along_z_type),
-            source=mpi_construct.previous_grid_along[z_axis],
+        self.comm_requests.append(
+            self.mpi_construct.grid.Irecv(
+                (local_field, self.recv_previous_along_z_type),
+                source=self.mpi_construct.previous_grid_along[z_axis],
+            )
         )
 
+    def exchange_vector_field_init(self, local_vector_field):
+        self.exchange_scalar_field_init(local_field=local_vector_field[0])
+        self.exchange_scalar_field_init(local_field=local_vector_field[1])
+        self.exchange_scalar_field_init(local_field=local_vector_field[2])
```

Comment:
same as above.

Reply:
Fixed in [72cf577](https://github.com/fankiat/sopht-mpi/pull/141/commits/72cf577bf8b80e2e8bfcb32872c9193350eeb215)